app/uk/gov/hmrc/cataloguefrontend/service/CostEstimationService.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cataloguefrontend.service
+
+import play.api.libs.json.{Json, Reads}
+import uk.gov.hmrc.cataloguefrontend.connector.ConfigConnector
+import uk.gov.hmrc.cataloguefrontend.model.Environment
+import uk.gov.hmrc.cataloguefrontend.service.CostEstimationService.{CostEstimation, DeploymentConfig}
+import uk.gov.hmrc.http.HeaderCarrier
+
+import java.util.Locale
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+final class CostEstimationService @Inject() (configConnector: ConfigConnector) {
+
+  def estimateServiceCost(
+    service: String,
+    environments: Seq[Environment]
+  )(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[CostEstimation] =
+    Future
+      .traverse(environments)(environment =>
```

Comment:
Here we're making a call to `service-configs` for each environment that the service runs in. Happy to revisit this and expose a new API in `service-configs` to fetch all configs for all environments if we'd prefer to do that though.